SmartDeviceLinkTests/DevAPISpecs/SDLPermissionFilterSpec.m

```diff
@@ -2,148 +2,228 @@
+                    __block NSDictionary<SDLPermissionRPCName, SDLRPCPermissionStatus *> *testObserverChangedDict = nil;
+                    __block SDLRPCPermissionStatus *rpcPermissionStatus1 = nil;
+                    __block SDLRPCPermissionStatus *rpcPermissionStatus2 = nil;
+                    __block SDLPermissionGroupStatus testObserverGroupStatus = SDLPermissionGroupStatusUnknown;
+
+                    beforeEach(^{
+                        rpcPermissionStatus1 = [[SDLRPCPermissionStatus alloc] initWithRPCName:testRPCName1 isRPCAllowed:YES rpcParameters:nil];
+                        rpcPermissionStatus2 = [[SDLRPCPermissionStatus alloc] initWithRPCName:testRPCName2 isRPCAllowed:NO rpcParameters:nil];
+                        testObserverChangedDict = @{testRPCName1: rpcPermissionStatus1,
+                                                    testRPCName2: rpcPermissionStatus2};
+                        testObserverGroupStatus = SDLPermissionGroupStatusMixed;
+
+                        testFilter.rpcPermissionStatusHandler(testObserverChangedDict, testObserverGroupStatus);
+                    });
+
+                    it(@"should call the changedDict correctly", ^{
+                        expect(testRPCPermissionStatusReturnChangedDict).to(equal(testObserverChangedDict));
+                    });
+
+                    it(@"should call the status correctly", ^{
+                        expect(@(testObserverGroupStatus)).to(equal(@(testObserverGroupStatus)));
+                    });
                 });
             });
         });
     });
-    
+
     describe(@"copying a filter", ^{
         __block SDLPermissionFilter *testFilter = nil;
         __block SDLPermissionFilter *testCopiedFilter = nil;
-        
+
         beforeEach(^{
-            testFilter = [SDLPermissionFilter filterWithRPCNames:@[testRPCName1] groupType:SDLPermissionGroupTypeAny observer:^(NSDictionary<SDLPermissionRPCName,NSNumber<SDLBool> *> * _Nonnull changedDict, SDLPermissionGroupStatus status) {}];
+            testFilter = [SDLPermissionFilter filterWithRPCNames:@[testPermissionElement1] groupType:SDLPermissionGroupTypeAny permissionsHandler:^(NSDictionary<SDLPermissionElement *,NSNumber<SDLBool> *> * _Nonnull changedDict, SDLPermissionGroupStatus status) {} rpcPermissionStatusHandler:nil];
             testCopiedFilter = [testFilter copy];
         });
-        
+
         it(@"should say copied filters are not the same instance", ^{
             expect(testCopiedFilter).toNot(beIdenticalTo(testFilter));
         });
-        
+
         it(@"should copy the identifier correctly", ^{
             expect(testCopiedFilter.identifier).to(equal(testFilter.identifier));
         });
-        
+
         it(@"should copy the filter array correctly", ^{
-            expect(testCopiedFilter.rpcNames).to(equal(testFilter.rpcNames));
+            expect(testCopiedFilter.permissionElements).to(equal(testFilter.permissionElements));
```

Comment:
I think this should fail if done correctly if you copy because it's checking references, which should be different. That means that this test is written incorrectly and succeeds incorrectly? The previous version would succeed because the strings were the same? I'm not sure.